### PR TITLE
Home

### PR DESCRIPTION
**Pull request by anyahrokhola**

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -24,6 +26,8 @@ describe('AppComponent', () => {
 			FormModule,
 			FormsModule,
 			ReactiveFormsModule,
+			ModalModule,
+			MatDialogModule, // todo remove matdialog
```
Comment (mishahrokhola) [inline src/app/app.component.spec.ts]: TODO?

File: src/app/modules/home/components/intro/intro.component.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class="intro-wrapper" style="background-image: url('../../../../../assets/intro-image.jpg')">
```
Comment (mishahrokhola) [inline src/app/modules/home/components/intro/intro.component.html]: fix path

File: src/app/modules/home/components/intro/intro.component.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
+@import '../../../../../assets/styles/typography.scss';
```
Comment (mishahrokhola) [inline src/app/modules/home/components/intro/intro.component.scss]: fix path

File: src/app/modules/home/components/intro/intro.component.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+@import '../../../../../assets/styles/button.scss';
+
+
+:host {
+	font-family: 'Roboto';
```
Comment (mishahrokhola) [inline src/app/modules/home/components/intro/intro.component.scss]: this is strange. This should be inside some text class or for specific heading tag like h1/h2 etc

File: src/app/modules/home/components/intro/intro.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { IntroComponent } from './intro.component';
+
+describe('IntroComponent', () => {
```
Comment (mishahrokhola) [inline src/app/modules/home/components/intro/intro.component.spec.ts:5]: spectator

File: src/app/modules/modal/components/modal-main/modal-main.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { ModalMainComponent } from './modal-main.component';
+
+describe('ModalMainComponent', () => {
```
Comment (mishahrokhola) [inline src/app/modules/modal/components/modal-main/modal-main.component.spec.ts:5]: spectator

File: src/app/modules/pages/components/footer/footer.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { FooterComponent } from './footer.component';
+
+describe('FooterComponent', () => {
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/footer/footer.component.spec.ts:6]: spectator

File: src/app/modules/pages/components/footer/footer.component.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.scss']
+})
+export class FooterComponent implements OnInit {
+
+  constructor() { }
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/footer/footer.component.ts:10]: remove empty constructor and methods

File: src/app/modules/pages/components/header/header.component.html [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<div class="container">
+	<div class="row align-items-center">
+		<div class="col-8">
+			<div class="logo-wrapper">
+				<a href="/"><img src="../../../../../assets/logo.png" alt="" /></a>
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/header/header.component.html]: fix path

File: src/app/modules/pages/components/header/header.component.scss [hunk 10]
```diff
@@ -0,0 +1,51 @@
+$color: #6e56fb;
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/header/header.component.scss:1]: should be in global variables and need a better name

File: src/app/modules/pages/components/header/header.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { HeaderComponent } from './header.component';
+
+describe('HeaderComponent', () => {
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/header/header.component.spec.ts:5]: spectator

File: src/app/modules/pages/components/social-media/social-media.component.html [hunk 12]
```diff
@@ -0,0 +1,21 @@
+<a href="https://twitter.com/?lang=ru">
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/social-media/social-media.component.html:1]: lang=ru? 
Zradonka

File: src/app/modules/pages/components/social-media/social-media.component.html [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+	</svg>
+</a>
+<a href="https://www.facebook.com/">
+	<svg xmlns="http://www.w3.org/2000/svg"  viewBox="0 0 24 24">
+		<path
+			d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm3 8h-1.35c-.538 0-.65.221-.65.778v1.222h2l-.209 2h-1.791v7h-3v-7h-2v-2h2v-2.308c0-1.769.931-2.692 3.029-2.692h1.971v3z"
+		/>
+	</svg>
+</a>
+<a href="https://www.instagram.com/?hl=ru">
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/social-media/social-media.component.html:15]: zradonka

File: src/app/modules/pages/components/social-media/social-media.component.scss [hunk 14]
```diff
@@ -0,0 +1,33 @@
+@import url('../../../../../assets/styles/typography.scss');
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/social-media/social-media.component.scss]: fix path

File: src/app/modules/pages/components/social-media/social-media.component.scss [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+
+$sm: 576px;
+$md: 768px;
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/social-media/social-media.component.scss]: remove duplicated variables. Import them from global variables

File: src/app/modules/pages/components/social-media/social-media.component.spec.ts [hunk 16]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { SocialMediaComponent } from './social-media.component';
+
+describe('SocialMediaComponent', () => {
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/social-media/social-media.component.spec.ts:5]: spectator

File: src/app/modules/pages/components/social-media/social-media.component.ts [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'social-media',
+  templateUrl: './social-media.component.html',
+  styleUrls: ['./social-media.component.scss']
+})
+export class SocialMediaComponent implements OnInit {
+
+  constructor() { }
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/social-media/social-media.component.ts]: remove empty method and constructor

File: src/assets/styles/typography.scss [hunk 18]
```diff
@@ -0,0 +1,51 @@
+$sm: 576px;
+$md: 768px;
+$lg: 992px;
```
Comment (mishahrokhola) [inline src/assets/styles/typography.scss]: import this variables from other file

File: src/assets/styles/variables.scss [hunk 19]
```diff
@@ -0,0 +1,4 @@
+$sm: 576px;
```
Comment (mishahrokhola) [inline src/assets/styles/variables.scss]: rename the file to media-variables

File: src/index.html [hunk 20]
```diff
@@ -9,6 +9,13 @@
 		<link rel="preconnect" href="https://fonts.gstatic.com" />
 		<link href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap" rel="stylesheet" />
 		<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
+
+		<link
+			href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.0/dist/css/bootstrap.min.css"
```
Comment (mishahrokhola) [inline src/index.html:14]: import only grid part of bootstrap, not whole framework

**Review by mishahrokhola — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts [hunk 21]
```diff
@@ -19,7 +22,7 @@ export class AppComponent implements OnInit {
 		test: new FormControl(),
 	});
 
-	constructor(private optionService: OptionService) {}
+	constructor(private optionService: OptionService, private dialog: MatDialog) {}
```
Comment (mishahrokhola) [inline src/app/app.component.ts]: create modal.module with custom modal.service that will use MatDialog and have method open

File: package.json [hunk 22]
```diff
@@ -22,6 +22,7 @@
 		"@angular/platform-browser": "~13.0.0",
 		"@angular/platform-browser-dynamic": "~13.0.0",
 		"@angular/router": "~13.0.0",
+		"ngx-bootstrap": "^8.0.0",
```
Comment (mishahrokhola) [inline package.json]: do we need this?

File: src/app/modules/pages/components/footer/footer.component.scss [hunk 23]
```diff
@@ -0,0 +1,30 @@
+$sm: 576px;
+$md: 768px;
+$lg: 992px;
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/footer/footer.component.scss]: import variables

File: src/app/modules/pages/components/footer/footer.component.scss [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+$xl: 1200px;
+
+:host{
+    background-color: #333;
```
Comment (mishahrokhola) [inline src/app/modules/pages/components/footer/footer.component.scss]: add variables for main colors

**Review by mishahrokhola — CHANGES_REQUESTED** [review]

File: src/app/modules/form/components/input-text/input-text.component.ts [hunk 25]
```diff
@@ -25,9 +27,19 @@ export class InputTextComponent implements ControlValueAccessor {
...
-		this.val = $MouseEvent.target.value;
-		this.onChange(this.val);
+	public setDisabledState(isDisabled: boolean): void {
+		this.disabled = isDisabled;
+	}
+	public handleInput(event: Event): void {
+		if (event.target instanceof HTMLInputElement) {
+			this.val = event.target.value;
+			this.onChange(this.val);
+			// eslint-disable-next-line no-console
```
Comment (mishahrokhola) [inline src/app/modules/form/components/input-text/input-text.component.ts:37]: remove console.log

File: src/app/modules/home/components/intro/intro.component.scss [hunk 26]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	@media screen and (max-width: $sm) {
+		font-size: 16px;
+	}
+}
+
+
+
+
```
Comment (mishahrokhola) [inline src/app/modules/home/components/intro/intro.component.scss:68]: remove extra empty space

File: src/app/modules/modal/services/modal.service.spec.ts [hunk 27]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { ModalService } from './modal.service';
+
+describe('ModalService', () => {
+	let service: ModalService;
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({});
```
Comment (mishahrokhola) [inline src/app/modules/modal/services/modal.service.spec.ts]: spectator

File: src/app/modules/modal/services/modal.service.ts [hunk 28]
```diff
@@ -0,0 +1,18 @@
...
+import { InputTextComponent } from '../../form/components/input-text/input-text.component';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class ModalService {
+	constructor(private dialog: MatDialog) {}
+
+	public openDialog() {
+		// eslint-disable-next-line no-console
```
Comment (mishahrokhola) [inline src/app/modules/modal/services/modal.service.ts:12]: You have no console here, so remove // eslint-disable-next-line no-console

File: src/app/modules/modal/services/modal.service.ts [hunk 29]
```diff
@@ -0,0 +1,18 @@
...
+		this.dialog.open(InputTextComponent, {
+			height: '400px',
+			width: '600px',
```
Comment (mishahrokhola) [inline src/app/modules/modal/services/modal.service.ts:15]: add config with this sizes

File: src/app/modules/button/directives/btn-intro/btn-intro.directive.spec.ts [hunk 30]
```diff
@@ -0,0 +1,22 @@
...
+	beforeEach(() => {
+		spectator = createDirective(`<button btnIntro>Text</button>`);
+	});
+
+	it('should create', () => {
+		expect(spectator.directive).toBeTruthy();
+	});
+
+	it('should add classes', () => {
+		expect(spectator.element.classList.toString()).toContain('button button-outline button-intro');
```
Comment (mishahrokhola) [inline src/app/modules/button/directives/btn-intro/btn-intro.directive.spec.ts:20]: expect(spectator.element).toHaveClass('button');
expect(spectator.element).toHaveClass('button-outline');
expect(spectator.element).toHaveClass('button-intro');

File: src/index.html (hunk 20 above)
Comment (anyahrokhola) [reply src/index.html:14]: fixed on the router branch

**Pull request merged**